### ISSUE TITLE
replace S3 and DynamoDB with opensource counterparts?

### ISSUE THREAD

**Issue by maxadamo** [camptocamp/terraboard#121]
Hi there! 
I was going through the documentation, and I had the feeling that we're getting back to those days, when Microsoft was dictating his rules.
S3, DynamoDB are proprietary and closed source technologies. AWS has never defined a RfC, or tried to create standards or try to reveal their source code and to me it's clear, that if we deal with opensource, we must promote opensource. 
Furthermore, AWS is not standard de-facto: I can be on premises, I can have google cloud, and even if, in my case I have AWS (and I really have it) I still don't understand the reasons behind this decision. 
Why not using Consul for the backend (which support the session locking)? Why not try something else for the versioning?

**Comment by raphink** [comment]
Hello,

You're welcome to contribute a PR to support consul, though our plan on the long term will be to store the states internally in Terraboard, making in a standard remote backend (like we did in TerraDB: https://github.com/camptocamp/terradb)

**Comment by hoshsadiq** [comment]
@maxadamo I've been running with Minio fine. Also, GCP, Azure, Gitlab and TFE/TFC support were added.

**Comment by maxadamo** [comment]
@hoshsadiq , it is true that Terraform supports several types of backends (and I have also noticed that Gitlab has added support to terraform), but not every backend support session locking, and it's not cleat to me if Terraboard is using `session locking`. I understand from @raphink that it has to be practical, and not too difficult to support in the tests **(*)**. 
**my personal opinion**
At first glance, a suitable backend seems to be `etcdv3`, because:
-  it can be embedded in the code (see Sensu Go 6.x), without having to setup and an external DB. 
- it can run on a single node instance, or clustered with 3 nodes. 

**(*)** for instance,I see that Postgres was added, and while it is a great Opensource software, it is probably overkill for the purpose, hard to maintain and it is only sensible, if you already have an existing database.
